### DISCUSSION TITLE
Forcing JUCE dialog and file window via LUA on pannel load

### DISCUSSION THREAD

**Discussion by damiensellier** [RomanKubiak/ctrlr#504] [Q&A] [answered]
Hi everybody,
We talked earlier about unchecking the Ctrlr property "Use OS native dialog windows" and "Use OS Native file open/save dialogs" for macOS BigSur and above.

![Native dialog settings](https://user-images.githubusercontent.com/55164385/202354658-95d1a93f-f107-49c3-9833-e2d3520fc752.png)

In Ctrlr core those property are called : ctrlrNativeAlerts and ctrlrNativeFileDialogs

Is there a way to force it (true, false) when loading a panel with a command like this: 
```

panel:getPanelEditor():setProperty("ctrlrNativeAlerts",false,true)
panel:getPanelEditor():setProperty("ctrlrNativeFileDialogs",false,true)
```

or 
```
panel:setProperty("ctrlrNativeAlerts",false,true)
panel:setProperty("ctrlrNativeFileDialogs",false,true)
```

My arguments are probably wrong because I don't get the desired effect but that would be so nice to be able to force it on loading to avoid those file management bugs with newer macOS.

Thanks !

Damien

**Comment by damiensellier** [comment, marked as answer]
The corresponding JUCE class for the File Manager is here :
https://docs.juce.com/master/classFileChooser.html

And is called via LuaBinds:
https://github.com/RomanKubiak/ctrlr/blob/c1f67d2f4033cb9f5dd20a123854fcddcd99b81d/Source/Lua/CtrlrLuaUtils.cpp#L146



I found something when calling an open/save window with :
utils.openFileWindow("Load preset from disk", File(""), "*.syx", **false**)

for exemple : 

`SysexFileFromDisk = utils.openFileWindow("Load preset from disk", File(""), "*.syx", false)`


The last bool argument is for the file manager type. 
_true_ is Native OS file window
_false_ is JUCE (ugly but working) file window

It's not exactly what I was looking for but it's a good fallback

**Comment by dnaldoog** [comment]
> panel:getPanelEditor():setProperty("ctrlrNativeAlerts",false,true)
> panel:getPanelEditor():setProperty("ctrlrNativeFileDialogs",false,true)

Hi @damien!

Whatever it is I bet it is:

<pre>
panel:getPanelEditor():setProperty("ctrlrNativeAlerts",1,true) -- 1/0
panel:getPanelEditor():setProperty("ctrlrNativeFileDialogs",0,true) -- 1/0
</pre>

... etc. 🐱

**Reply by damiensellier** [reply]
Hi John, 1/0 makes more sense than true/false.
Anyway, since "useOSNativeDialogBox" variable is required and sets the dialog type to use with utils.openFileWindow(), my initial question is useless. 
The only real way to make it work from scratch for BigSur would be to force the default value to "false" in "CtrlrManager.cpp" at the build stage for CTRLR file manager windows like "open/save panel" from the menu. 

https://github.com/RomanKubiak/ctrlr/blob/c1f67d2f4033cb9f5dd20a123854fcddcd99b81d/Source/Core/CtrlrManager/CtrlrManager.cpp#L80


I'll do it for my next Ctrlr build to be sure it's working for BigSur unless the problem is no longer there in the latest version Monterey or Ventura.
